WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/AdaptiveStack.swift

```diff
@@ -0,0 +1,31 @@
+import SwiftUI
+
+struct AdaptiveStack<Content: View>: View {
+    @Environment(\.sizeCategory) var sizeCategory
+    @Environment(\.horizontalSizeClass) var horizontalSizeClass
+
+    let horizontalAlignment: HorizontalAlignment
+    let verticalAlignment: VerticalAlignment
+    let spacing: CGFloat?
+    let content: () -> Content
+
+    init(horizontalAlignment: HorizontalAlignment = .center,
+         verticalAlignment: VerticalAlignment = .center,
+         spacing: CGFloat? = nil,
+         @ViewBuilder content: @escaping () -> Content) {
+        self.horizontalAlignment = horizontalAlignment
+        self.verticalAlignment = verticalAlignment
+        self.spacing = spacing
+        self.content = content
+    }
+
+    var body: some View {
+        Group {
+            if sizeCategory.isAccessibilityCategory, horizontalSizeClass == .compact {
```

Comment:
I noticed in the [documentation for `isAccessibilityCategory`](https://developer.apple.com/documentation/swiftui/contentsizecategory/isaccessibilitycategory) that it's being deprecated. Do you think it's worth switching to something like this instead:

```suggestion
            if sizeCategory >= .accessibilityMedium, horizontalSizeClass == .compact {
```

Reply:
Good find! it's actually a deprecation for whole [ContentSizeCategory](https://developer.apple.com/documentation/swiftui/contentsizecategory) enum. It's replaced with [DynamicTypeSize](https://developer.apple.com/documentation/swiftui/dynamictypesize) on iOS 15.

I don't see obvious solution to conditionally add `@Environment` property only for iOS 15. Let's leave it for now and trust Xcode to remind us about migration in the future 🙂